### ISSUE TITLE
Question regarding logits = sum_of_squares - z.shape[1]

### ISSUE THREAD

**Issue by YangLiangwei** [loeweX/Forward-Forward#8]
On file src.ff_model.py line 76:
        logits = sum_of_squares - z.shape[1]
        ff_loss = self.ff_loss(logits, labels.float())

I wondered for a long time and did not figure out why you wrote (sum_of_squares - z.shape[1]). Can you clarify it for me?
Besides, is the implementation of ff_loss the same as Hinton's original paper? From my understanding, it is not the same, and why do you make this modification?

**Comment by whubaichuan** [comment]
@YangLiangwei I think (sum_of_squares - z.shape[1]) is based on the Equation (1) of Hinton's FF paper.
